b2sdk/file_lock.py

```diff
@@ -135,42 +170,75 @@ def add_to_to_upload_headers(self, headers):
             self.mode.value
         )  # mode = NONE is not supported by the server at the
         # moment, but it should be
-        headers['X-Bz-File-Retention-Retain-Until-Timestamp'] = self.retain_until
+        headers['X-Bz-File-Retention-Retain-Until-Timestamp'] = str(self.retain_until)
+
+    def __eq__(self, other):
+        if not isinstance(other, FileRetentionSetting):
+            return NotImplemented
+        return self.mode == other.mode and self.retain_until == other.retain_until
+
+    def __repr__(self):
+        return '%s(%s, %s)' % (
+            self.__class__.__name__, repr(self.mode.value), repr(self.retain_until)
+        )
+
+
+@enum.unique
+class LegalHold(enum.Enum):
+    """Information about legalHold switch in a file."""
+
+    ON = 'on'
+    OFF = 'off'
+    UNSET = None  # this is the server default, as for now it is functionally equivalent to OFF
+    UNKNOWN = 'unknown'  # This one is used if the client is not authorized to read retention settings
 
+    def __bool__(self):
+        if self is LegalHold.UNKNOWN:
```

Comment:
this is three value logic and unknown is valid value... it should not raise an error:
```
if file_version.legal_hold:
    ...
```
???

Reply:
I would add:
```
def is_on(self): return self is LegalHold.ON
def is_off(self): return self is LegalHold.OFF or self is LegalHold.UNSET
def is_unknown(self): return self is LegalHold.UNKNOWN
```

Reply:
so then:
```
if file_version.legal_hold.is_on():
    ...
```

Reply:
I think you're right. This was thought through, but It is over-engineered